core/src/main/java/com/distkv/dst/core/concepts/DstSortedListsImpl.java

```diff
@@ -1,64 +1,49 @@
 package com.distkv.dst.core.concepts;
 
 import com.distkv.dst.common.exception.KeyNotFoundException;
+import com.distkv.dst.common.exception.SortedListDuplicatedMembersException;
+import com.distkv.dst.common.exception.SortedListIncrScoreOutOfRange;
 import com.distkv.dst.common.exception.SortedListMemberNotFoundException;
-import com.distkv.dst.common.exception.SortedListTopNumBePositiveException;
+import com.distkv.dst.common.exception.SortedListTopNumBeNonNegativeException;
 import com.distkv.dst.common.entity.sortedList.SortedListEntity;
+import com.distkv.dst.core.struct.slist.SortedList;
+import com.distkv.dst.core.struct.slist.SortedListLinkedListImpl;
 
 import java.util.List;
 import java.util.LinkedList;
-import java.util.Collections;
-import java.util.ListIterator;
-import java.util.Arrays;
 
-public class DstSortedListsImpl extends DstConcepts<LinkedList<SortedListEntity>>
+public class DstSortedListsImpl
+    extends DstConcepts<SortedList>
     implements DstSortedLists {
-
   public DstSortedListsImpl() {
+
   }
 
   @Override
   public void put(String key, LinkedList<SortedListEntity> list) {
-    // Note(qwang): Overwrite for do a sort.
-    Collections.sort(list);
-    dstKeyValueMap.put(key, list);
+    SortedList sortedList = new SortedListLinkedListImpl();
+    if (!sortedList.put(list)) {
```

Comment:
You mean we could put one key twice?

Reply:
Sorry, I did not take this into account. I will complete the method.